File: solutions/29ShuffleAnArray/java/ShuffleAnArray.java
class ShuffleAnArray {
  private int[] nums;

  public ShuffleAnArray(int[] nums) {
    this.nums = nums;
  }

  public int[] reset() {
    return nums;
  }

  public int[] shuffle() {
    int[] a = nums.clone();
    for (int i = 0; i < a.length; i++) {
      int j = (int) (Math.random() * a.length);
      int temp = a[i];
      a[i] = a[j];
      a[j] = temp;
    }
    return a;
  }
}

File: solutions/29ShuffleAnArray/python/ShuffleAnArray.py
import random


class Solution(object):
    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.nums = nums
        self.original = list(nums)

    def reset(self):
        """
        Resets the array to its original configuration and return it.
        :rtype: List[int]
        """
        self.nums = list(self.original)
        return self.nums

    def shuffle(self):
        """
        Returns a random shuffling of the array.
        :rtype: List[int]
        """
        random.shuffle(self.nums)
        return self.nums
